dev-python/spyder/spyder-4.0.1.ebuild

```diff
@@ -0,0 +1,134 @@
+	dev-python/pickleshare[${PYTHON_USEDEP}]
+	dev-python/psutil[${PYTHON_USEDEP}]
+	>=dev-python/pygments-2.0[${PYTHON_USEDEP}]
+	dev-python/pylint[${PYTHON_USEDEP}]
+	>=dev-python/python-language-server-0.31.2[${PYTHON_USEDEP}]
+	>=dev-python/pyxdg-0.26[${PYTHON_USEDEP}]
+	dev-python/pyzmq[${PYTHON_USEDEP}]
+	>=dev-python/qdarkstyle-2.7[${PYTHON_USEDEP}]
+	>=dev-python/qtawesome-0.5.7[${PYTHON_USEDEP}]
+	>=dev-python/qtconsole-4.6.0[${PYTHON_USEDEP}]
+	>=dev-python/QtPy-1.5.0[${PYTHON_USEDEP},svg,webengine]
+	dev-python/sphinx[${PYTHON_USEDEP}]
+	>=dev-python/spyder-kernels-1.8.1[${PYTHON_USEDEP}]
+	dev-python/watchdog[${PYTHON_USEDEP}]"
+
+DEPEND="test? (
+	dev-python/coverage[${PYTHON_USEDEP}]
+	dev-python/cython[${PYTHON_USEDEP}]
+	dev-python/flaky[${PYTHON_USEDEP}]
+	dev-python/matplotlib[tk,${PYTHON_USEDEP}]
+	dev-python/mock[${PYTHON_USEDEP}]
+	dev-python/pandas[${PYTHON_USEDEP}]
+	dev-python/pillow[${PYTHON_USEDEP}]
+	dev-python/pytest-lazy-fixture[${PYTHON_USEDEP}]
+	dev-python/pytest-mock[${PYTHON_USEDEP}]
+	dev-python/pytest-qt[${PYTHON_USEDEP}]
+	sci-libs/scipy[${PYTHON_USEDEP}]
+	dev-python/sympy[${PYTHON_USEDEP}] )"
+
+# Based on the courtesy of Arfrever
+# This patch removes a call to update-desktop-database during build
+# This fails because access is denied to this command during build
+PATCHES=( "${FILESDIR}"/${P}-build.patch )
+
+distutils_enable_tests pytest
+
+python_prepare_all() {
+	# skip uri (online) tests
+	rm spyder/plugins/editor/widgets/tests/test_goto.py || die
+
+	# skip online test
+	rm spyder/widgets/github/tests/test_github_backend.py
+
+	# Assertion error, looks like an online test
+	rm spyder/utils/tests/test_vcs.py || die
+
+		# No idea why this fails, no error just stops and dumps core
+	sed -i -e 's:test_arrayeditor_edit_complex_array:_&:' \
+		spyder/plugins/variableexplorer/widgets/tests/test_arrayeditor.py || die
+
+	# Assertion error, can't connect/remember inside ebuild environment
+	sed -i -e 's:test_connection_dialog_remembers_input_with_password:_&:' \
+		spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py || die
+	sed -i -e 's:test_connection_dialog_remembers_input_with_ssh_passphrase:_&:' \
+		spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py || die
+
+	# Assertion error, no idea why
+	sed -i -e 's:test_pdb_multiline:_&:' \
+		spyder/plugins/ipythonconsole/tests/test_ipythonconsole.py || die
+
+	# Assertion error (pytest-qt), maybe we can't do shortcuts inside ebuild environment?
+	sed -i -e 's:test_transform_to_uppercase_shortcut:_&:' \
+		spyder/plugins/editor/widgets/tests/test_shortcuts.py || die
+	sed -i -e 's:test_transform_to_lowercase_shortcut:_&:' \
+		spyder/plugins/editor/widgets/tests/test_shortcuts.py || die
+	sed -i -e 's:test_go_to_line_shortcut:_&:' \
+		spyder/plugins/editor/widgets/tests/test_shortcuts.py || die
+	sed -i -e 's:test_delete_line_shortcut:_&:' \
+		spyder/plugins/editor/widgets/tests/test_shortcuts.py || die
```

Comment:
Make this into a one sed call please. 

There are multiple separate seds, should these be made into patch/patches instead?

Reply:
I was using sed here, because patches tend to break on version bumps. I version bumped some other packages earlier, and the patches designed to skip specific tests all broke. Also it is a lot easier to write/change a sed line then it is to adjust a patch. So unless there is a good reason to use a patch, I'd prefer to use sed.

Reply:
That is the reason. These patches should be reported and fixed upstream as much as possible. Having these sed lines make it so comfortable that after couple of bumps, you wont remember they are there. Then in some future bump they might actually break something without the maintainer noticing.